### ISSUE TITLE
Search is broken [ed: Does not find NMSSH pod]

### ISSUE THREAD

**Issue by ReDetection** [CocoaPods/cocoapods.org#213]
I had an idea for an app, and app intended to use ssh connection, so I opened cocoapods.org and searched for SSH and got only one result (SSHTTPClient, definitely not what I was looking for). So I managed to search on the github and found two useful repos: [a script to compile libssh2 for iOS](https://github.com/x2on/libssh2-for-iOS) and [Objective-C libssh OS X wrapper](https://github.com/codinn/SSHKitCore). Unfortunately, I wasn't able to quickly plug them in, but I'm not the person who give up quickly, so I spent about a week of my spare time to compile them. You can imagine, how surprised I was when I accidentally find [this up-to-date useful pod](https://github.com/Lejdborg/NMSSH), and [this](https://github.com/libssh2/libssh2), and some others. 

**TL;DR:** 

search on the cocoapods.org is broken and can show completely different results than `pod search`.  

What can be the issue? Too short query? Stopwords? Why `pod install` works fine then?

**Comment by floere** [comment]
Also try making the title more descriptive please (would also make it more memorable). Thanks! 😊

**Comment by ReDetection** [comment]
@floere done! again, sorry for the accidental empty issue.

**Comment by floere** [comment]
Thanks for the details! I've added some relevant info to the title.

**Comment by floere** [comment]
That "some others" list of pods you mention would be very helpful for us. Can you supply it?

**Comment by ReDetection** [comment]
Sure. There is only one more pod related to ssh: https://github.com/endSly/libssh2-pod. Others are just unrelated pods but they have ssh as a substring (mostly a lot of progresshud implementations). 
I've made a gist with full `pod search ssh` output here https://gist.github.com/ReDetection/ebac402bc5be61847675

**Comment by floere** [comment]
Thanks for gisting the output! Looks like pod search returns mostly pods which have nothing to do with SSH due to pod search just searching for it as a substring.

Ok, so the reason for the two ssh pods not being found is that we only find full words in the summary, such as SSH2 (https://cocoapods.org/?q=SSH2).

Thanks for your feedback!
I believe the way to correct this is by tagging these pods - in the search engine indexing process - with the SSH tag.

* [ ] Also tag SSH2 related pods with SSH tag.

**Comment by orta** [comment]
> I believe the way to correct this is by tagging these pods - in the search engine indexing process - with the SSH tag.

:+1:

**Comment by ReDetection** [comment]
If you search only for the full words in summary, how the `SSHTTPClient` showed up then?
![screen shot 2015-09-26 at 19 48 05](https://cloud.githubusercontent.com/assets/458005/10117858/6f5a2bfe-647f-11e5-8716-0675bbfb355b.png)

**Comment by floere** [comment]
It's because we also search in the pod name - according to some rules.

**Comment by ReDetection** [comment]
So `NMSSH` have an SSH substring there. Why it's not being found?

**Comment by floere** [comment]
Here's how we split the name for indexing: https://github.com/CocoaPods/search.cocoapods.org/blob/master/lib/models/pod.rb#L347 These words are then prefix searchable.

**Comment by floere** [comment]
Here's how we split the name for indexing: https://github.com/CocoaPods/search.cocoapods.org/blob/master/lib/models/pod.rb#L347 These words are then prefix searchable.

**Comment by ReDetection** [comment]
It's cool for the ranging to move more relevant stuff to the top, but shouldn't the all pods with query substring to be found?

**Comment by floere** [comment]
Not generally, no – because you'd get far too many irrelevant results (like with pod search).

As always, a balancing act.

**Comment by ReDetection** [comment]
Disagree because of three points:
• omitted results did contain exactly what I was looking for
• something is better than nothing
• I'm arguing to do this name splitting for the ranging purposes, to offer more relevant content first, but for sure there should be all the results displayed. Just imagine if google will omit 90% of it's output.

**Comment by floere** [comment]
On cocoapods.org we value precision more than recall (while continuously aiming to improve recall without detriment to precision), while with `pod search` the focus is fully on recall.

Your feedback has helped improve our engine (https://cocoapods.org/?q=ssh). Thanks for that! :rocket:

**Comment by ReDetection** [comment]
@floere I appreciate you have improved your engine and included more results there, but the most proper pod for the SSH are still not in that list. Can you please tune your engine once more and check NMSSH finds?

**Comment by floere** [comment]
@ReDetection It's there now. Feel free to also contact pod authors in case their summaries are not descriptive enough (the first pod, for example, only says "Sample network call.", which is not helpful) – we depend on these summaries to build the index.

**Comment by ReDetection** [comment]
@floere Thanks a lot!

**Comment by floere** [comment]
@ReDetection Cheers! :rocket:
